### PR TITLE
Update microblaze to work with -a -p

### PR DESCRIPTION
**Pull request by msteveb**
The new resolved-symbol mode works much better than the old way of invoking the linker 3 times, but the necessary support for this was missing for microblaze.

With these changes I have a crosstool-ng gcc 9.2.0 now building working BFLT executables.

### CODE REVIEW

File: elf2flt.c [hunk 1]
```diff
@@ -2036,5 +2037,5 @@ int main(int argc, char *argv[])
  * c-basic-offset: 4
  * tab-width: 8
  * end:
- * vi: tabstop=8 shiftwidth=4 textwidth=79 noexpandtab
+ * vi: tabstop=8 shiftwidth=4 noexpandtab
```
Comment (vapier) [inline elf2flt.c]: unrelated change ?

**Review by vapier — CHANGES_REQUESTED** [review]
i am by no means an microblaze expert.  the changes look reasonable to me in that context.

if no one has any feedback on the topic, i'm fine to merge it in like a week and wait for someone to complain :).

File: elf2flt.c (hunk 1 above)
Comment (msteveb) [reply elf2flt.c]: Oops. Do you want me to force push a new version with this removed?

Comment (vapier) [reply elf2flt.c]: yeah, rewrite the commits so there are only the two "clean" ones

Comment (msteveb) [reply elf2flt.c]: Done!

**Comment by msteveb** [comment]
Testing with earlier gcc 8.2.0 and binutils 2.31 shows that we need to support an additional dummy reloc type, R_MICROBLAZE_32_NONE. I pushed one more more commit for this.

**Review by vapier — APPROVED** [review]

File: elf2flt.c [hunk 2]
```diff
@@ -66,6 +66,9 @@ const char *elf2flt_progname;
 #include <elf/m68k.h>
 #elif defined(TARGET_microblaze)
 #include <elf/microblaze.h>
+#if R_MICROBLAZE_max < 33
```
Comment (vapier) [inline elf2flt.c]: is this a define ?  isn't this just an enum ?  and thus this check would always be true (since undefined symbol is 0 which is less than 33) ?

Comment (msteveb) [reply elf2flt.c]: Yes, you are right! The good news is that that the #define will work even if it already exists as an enum, but the #if does nothing useful.

Comment (vapier) [reply elf2flt.c]: i'm sure it compiles OK, but it's still confusing/misleading code that i don't think we should merge.

also, should we be supporting non-standard relocs ?  if it's not merged into binutils itself, then there's no guarantee that reloc 33 will be R_MICROBLAZE_32_NONE.

Comment (msteveb) [reply elf2flt.c]: It's fine not to merge this. Maybe one day when it's merged into binutil mainline

File: elf2flt.c [hunk 3]
```diff
@@ -977,45 +999,20 @@ output_relocs (
...
-				}
+					r_mem[2] = (sym_addr >> 24) & 0xff;
+					r_mem[3] = (sym_addr >> 16) & 0xff;
+					r_mem[6] = (sym_addr >>  8) & 0xff;
+					r_mem[7] =  sym_addr        & 0xff;
+					relocation_needed = 1;
+					/* The symbol is split over two consecutive instructions.  
+					   Flag this to the flat loader by setting the high bit of 
+					   the relocation symbol. */
+					pflags=0x80000000;
```
Comment (vapier) [inline elf2flt.c]: i know you're just moving code around, but please add spaces around the `=` while you're here

File: elf2flt.c (hunk 2 above)
Comment (vapier) [reply elf2flt.c]: you could test for the define via a configure test, then we wouldn't hardcode the reloc number.  still not great as the reloc hasn't been upstreamed, but it's a bit more palatable, and we can add a comment explaining the origin.

we're doing that for new xtensa relocs in #18 if you want to see one way of doing it.

**Comment by msteveb** [comment]
I decided to just remove the R_MICROBLAZE_32_NONE change. Will keep it as a private patch until  it makes into mainline (if ever). Added whitespace as you requested.

**Comment by vapier** [comment]
hmm, i think the minor whitespace fix didn't come across ?

**Comment by msteveb** [comment]
No? Did I not get that right? I changed this: https://github.com/msteveb/elf2flt/blob/92f41a3b49631846bd306f575f5a003003f1ee76/elf2flt.c#L634

Perhaps you wanted me to change this instead: https://github.com/msteveb/elf2flt/blob/92f41a3b49631846bd306f575f5a003003f1ee76/elf2flt.c#L1018

**Comment by vapier** [comment]
the comment i left above was indeed on line 1018 ;)

**Comment by msteveb** [comment]
Darn! Do you want both fixed or just 1018 then?

**Comment by vapier** [comment]
update the commit you have now with 1018 fixed and we should be good to go

**Comment by msteveb** [comment]
done
